src/OrchardCore/OrchardCore.Infrastructure/Documents/DefaultDocumentSerializer.cs

```diff
@@ -14,13 +15,10 @@ public class DefaultDocumentSerializer : IDocumentSerializer
     {
         public static readonly DefaultDocumentSerializer Instance = new();
 
-        public DefaultDocumentSerializer()
+        public Task<byte[]> SerializeAsync<TDocument>(TDocument document, int compressThreshold = int.MaxValue)
+            where TDocument : class, IDocument, new()
         {
-        }
-
-        public Task<byte[]> SerializeAsync<TDocument>(TDocument document, int compressThreshold = int.MaxValue) where TDocument : class, IDocument, new()
-        {
-            var data = Encoding.UTF8.GetBytes(JConvert.SerializeObject(document));
+            var data = JsonSerializer.SerializeToUtf8Bytes(document, JOptions.Default);
```

Comment:
maybe we should since the document store by default uses `DocumentJsonSerializerOptions`.

Reply:
Done

Reply:
I actually think there is an unknown bug out there that we have not spotted yet because `DocumentJsonSerializerOptions` is not being used here. `DocumentJsonSerializerOptions` uses `PolymorphicJsonTypeInfoResolver` where the `JOptions.Default` does not. So I think any document that uses `$type` like won't be serialized/deserialized correctly from the distributed cache.

Reply:
Great, then!